x-pack/plugins/security_solution/scripts/endpoint/endpoint_agent_runner/vm_config/vm_config_cloud_init.yaml

```diff
@@ -0,0 +1,4 @@
+users:
+  - name: ubuntu
+    ssh_authorized_keys:
```

Comment:
adds the authorized SSH key to the VM

Reply:
Could the ssh key be generated on the fly so it don't needs to be on the repo?

Reply:
Hey @dasansol92 - thanks for the feedback. I have reverted the changes that use SSH to run the agent, so this is no longer part of the PR.